Content.Shared/Movement/Systems/MovementSpeedModifierSystem.cs

```diff
@@ -50,6 +50,28 @@ public void ChangeFriction(EntityUid uid, float friction, float? frictionNoInput
             move.Acceleration = acceleration;
             Dirty(uid, move);
         }
+
+        /// <summary>
+        /// Changes the way being weightless affects your movement.
+        /// Only use this if you want to change the behaviour without changing it back again.
+        /// </summary>
+        public void ChangeWeightless(
+            EntityUid uid,
+            float? weightlessModifier = null,
+            float? weightlessAcceleration = null,
+            float? weightlessFriction = null,
+            float? weightlessFrictionNoInput = null,
+            MovementSpeedModifierComponent? move = null)
+        {
+            if (!Resolve(uid, ref move, false))
+                return;
+
+            move.WeightlessModifier = weightlessModifier ?? move.WeightlessModifier;
+            move.WeightlessAcceleration = weightlessAcceleration ?? move.WeightlessAcceleration;
+            move.WeightlessFriction = weightlessFriction ?? move.WeightlessFriction;
+            move.WeightlessFrictionNoInput = weightlessFrictionNoInput ?? move.WeightlessFrictionNoInput;
+            Dirty(uid, move);
+        }
     }
```

Comment:
Please just make it a setter for 1 field, not 4.

Reply:
Sorry but could you please explain? My brain is lacking the thinking juices right now.

Reply:
Wait I think I figured it out never mind.

Reply:
Also why isn't this just using the event?